tasks/sep-dev-0/006-holocene-sys-cfg/SignFromJson.s.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {SignFromJson as OriginalSignFromJson} from "script/SignFromJson.s.sol";
+import {Simulation} from "@base-contracts/script/universal/Simulation.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {LibString} from "solady/utils/LibString.sol";
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import {DisputeGameFactory} from "@eth-optimism-bedrock/src/dispute/DisputeGameFactory.sol";
+import {FaultDisputeGame} from "@eth-optimism-bedrock/src/dispute/FaultDisputeGame.sol";
+import {PermissionedDisputeGame} from "@eth-optimism-bedrock/src/dispute/PermissionedDisputeGame.sol";
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {Proxy} from "@eth-optimism-bedrock/src/universal/Proxy.sol";
+
+contract NestedSignFromJson is OriginalSignFromJson {
+    using LibString for string;
+
+    // Safe contract for this task.
+    address immutable proxyAdminOwnerSafe = vm.envAddress("OWNER_SAFE");
+
+    address proxyAdmin;
+    SystemConfig systemConfig;
+    address systemConfigImpl;
+
+    address l1XDMProxy;
+    address l1ERC721BridgeProxy;
+    address l1StandardBridgeProxy;
+    address optimismPortalProxy;
+    address optimismMintableERC20FactoryProxy;
+    address batchInbox;
+    uint256 startBlock;
+    address dgfProxy;
+
+    /// @notice Sets up the dgfProxy
+    function setUp() public {
+        string memory inputJson;
+        string memory path = "/tasks/sep-dev-0/006-holocene-sys-cfg/input.json";
+        try vm.readFile(string.concat(vm.projectRoot(), path)) returns (string memory data) {
+            inputJson = data;
+        } catch {
+            revert(string.concat("Failed to read ", path));
+        }
+
+        string memory l2ChainId = "11155421";
+
+        proxyAdmin = readContractAddress("ProxyAdmin", l2ChainId);
+        systemConfigImpl = stdJson.readAddress(inputJson, "$.transactions[9].contractInputsValues._implementation");
+        systemConfig = SystemConfig(readContractAddress("SystemConfigProxy", l2ChainId));
+
+        l1XDMProxy = readContractAddress("L1CrossDomainMessengerProxy", l2ChainId);
+        l1ERC721BridgeProxy = readContractAddress("L1ERC721BridgeProxy", l2ChainId);
+        l1StandardBridgeProxy = readContractAddress("L1StandardBridgeProxy", l2ChainId);
+        optimismPortalProxy = readContractAddress("OptimismPortalProxy", l2ChainId);
+        optimismMintableERC20FactoryProxy = readContractAddress("OptimismMintableERC20FactoryProxy", l2ChainId);
+        batchInbox = stdJson.readAddress(inputJson, "$.transactions[6].contractInputsValues._addr");
```

Comment:
This is also in the superchain registry: https://github.com/ethereum-optimism/superchain-registry/blob/a560627d56d9b12c7c649e782110101a9579a2c4/superchain/configs/sepolia-dev-0/oplabs-devnet-0.toml#L8